sycl/include/CL/sycl/backend_types.hpp

```diff
@@ -10,12 +10,34 @@
 
 #include <CL/sycl/detail/defines.hpp>
 
+#include <iostream>
+
 __SYCL_INLINE_NAMESPACE(cl) {
 namespace sycl {
 
 enum class backend : char { host, opencl, level0, cuda };
 
 template <backend name, typename SYCLObjectT> struct interop;
 
+inline std::ostream &operator<<(std::ostream &Out, backend be) {
+  switch (be) {
+  case backend::host:
+    Out << std::string("host");
+    break;
+  case backend::opencl:
+    Out << std::string("opencl");
+    break;
+  case backend::level0:
+    Out << std::string("level0");
+    break;
+  case backend::cuda:
+    Out << std::string("cuda");
+    break;
+  default:
+    Out << std::string("unknown");
```

Comment:
I'm afraid, this can trigger `-Wcovered-switch-default` warning on newer compilers as you've already covered all of enum values: https://github.com/intel/llvm/blob/49b6223cde1e999231ef29f7cd737bf597a1851d/sycl/include/CL/sycl/backend_types.hpp#L16

Reply:
Done.